synapse/replication/tcp/handler.py

```diff
@@ -151,6 +151,13 @@ def start_replication(self, hs):
             hs.get_reactor().connectTCP(host, port, self._factory)
 
     async def on_REPLICATE(self, conn: AbstractConnection, cmd: ReplicateCommand):
+        self.send_positions_to_connection(conn)
+
+    def send_positions_to_connection(self, conn: AbstractConnection):
```

Comment:
This doesn't seem to use `conn` at all. Is that on purpose?

Reply:
Oh, that's exciting! It *should* only be sending stuff to the given connection via `conn.send_command` but instead is sending it to all connections via `self.send_command`. It doesn't actually matter that we send `POSITION` spuriously to random connections, but lets fix it